### PR TITLE
ui: Refactor UI config

### LINKED ISSUE #860: Refactor UI config
UI code currently directly refers to environment variables in multiple places. We should initialize config from env variables in one place, and then refer to that configuration wherever it is needed. Could also consider using https://env.t3.gg/ for typesafe env variables in addition.

### PR DESCRIPTION
**Pull request by Etsija**
Please see the individual commits for details.

Resolves #860.

### CODE REVIEW

File: ui/package.json [hunk 1]
```diff
@@ -32,6 +32,7 @@
     "@radix-ui/react-tabs": "^1.1.0",
     "@radix-ui/react-toast": "^1.1.5",
     "@radix-ui/react-tooltip": "^1.0.7",
+    "@t3-oss/env-core": "^0.11.1",
```
Comment (mnonnenmacher) [inline ui/package.json:35]: Could you shortly summarize the benefits of using T3 in the commit message?

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: ui/src/config.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
+/*
```
Comment (lamppu) [inline ui/src/config.ts:1]: Commit message typo: Inizialise -> Initialize

**Comment by Etsija** [comment]
I'd really appreciate an opinion from @mmurto as well to this PR.

File: ui/package.json (hunk 1 above)
Comment (sschuberth) [reply ui/package.json:35]: Commit message nit: There is no point in using `[1]` if you don't refer to it 😉 I propose to change this to:

```
T3 Env [1] ...

...

[1]: https://env.t3.gg/docs/core
```

So refer to the footnote in the first sentence, and drop "See " from the footnote.

**Review by mmurto — CHANGES_REQUESTED** [review]

**Review by mmurto — APPROVED** [review]

**Pull request merged**